libs/agno/agno/tools/evm.py

```diff
@@ -0,0 +1,96 @@
+"""EVM integration tools for interacting with EVM Blockchains."""
+
+from os import getenv
+
+from web3 import Web3
+
+from agno.tools import Toolkit
+from agno.utils.log import logger
+
+
+class EvmTools(Toolkit):
+    def __init__(
+        self,
+        private_key: str,
+        rpc_url: str,
+    ):
+        """Initialise EVM tools."""
+
+        super().__init__(name="evm_tools")
+
+        self.private_key = private_key or getenv("EVM_PRIVATE_KEY")
+        self.rpc_url = rpc_url or getenv("EVM_RPC_URL")
+
+        if not self.private_key:
+            logger.error("Private Key is required")
+            raise ValueError("Private Key is required")
+        if not self.rpc_url:
+            logger.error("RPC Url is needed to interact with EVM blockchain")
+            raise ValueError("RPC Url is needed to interact with EVM blockchain")
+        if not self.private_key.startswith("0x"):
+            self.private_key = f"0x{self.private_key}"
+        self.w3 = Web3(Web3.HTTPProvider(self.rpc_url))
+        self.account = self.w3.eth.account.from_key(self.private_key)
+        logger.info(f"Your wallet address is: {self.account.address} ")
+
+    def get_max_priority_fee_per_gas(self) -> int:
```

Comment:
Can this not just happen as part of the next function? Or does it need to be its own function

Reply:
Or is only the last function an actual tool for the agent?